packages/google_maps_flutter/google_maps_flutter/ios/Classes/GoogleMapController.m

```diff
@@ -119,9 +115,38 @@ - (instancetype)initWithFrame:(CGRect)frame
 }
 
 - (UIView*)view {
+  [_mapView addObserver:self forKeyPath:@"frame" options:0 context:nil];
   return _mapView;
 }
 
+- (void)observeValueForKeyPath:(NSString*)keyPath
+                      ofObject:(id)object
+                        change:(NSDictionary*)change
+                       context:(void*)context {
+  if (_cameraDidInitialSetup) {
+    // We only observe the frame for initial setup.
+    [_mapView removeObserver:self forKeyPath:@"frame"];
+    return;
+  }
+  if (object == _mapView && [keyPath isEqualToString:@"frame"]) {
+    CGRect bounds = _mapView.bounds;
+    if (CGRectEqualToRect(bounds, CGRectZero)) {
+      // Rerely, frame can change without actually changing the size of the view;
+      // eg, consider a frame change such as: (0, 0, 0, 0) -> (10, 10, 0, 0)
+      // We ignore this type of changes.
+      return;
+    }
+    _cameraDidInitialSetup = YES;
```

Comment:
Why not remove the observer here instead of on the next frame change?

Reply:
good point! Thanks. will change.